src/mock_crust/support.rs

```diff
@@ -407,34 +417,41 @@ impl<UID: Uid> ServiceImpl<UID> {
 
         let result = ConnectionInfoResult {
             result_token,
-            result: Ok(PrivConnectionInfo {
-                id: unwrap!(self.uid),
+            result: Ok(PubConnectionInfo {
+                id: unwrap!(self.uid.as_ref()).public_keys().clone(),
                 endpoint: self.endpoint,
             }),
         };
 
         self.send_event(CrustEvent::ConnectionInfoPrepared(result));
     }
 
-    pub fn connect(&self, _our_info: PrivConnectionInfo<UID>, their_info: PubConnectionInfo<UID>) {
-        let packet = Packet::ConnectRequest(unwrap!(self.uid), their_info.id);
+    pub fn connect(&self, _our_info: PubConnectionInfo, their_info: PubConnectionInfo) {
+        let packet = Packet::ConnectRequest(
+            unwrap!(self.uid.as_ref()).public_keys().clone(),
+            their_info.id,
+        );
         self.send_packet(their_info.endpoint, packet);
     }
 
     pub fn set_accept_bootstrap(&mut self, accept: bool) {
         self.accept_bootstrap = accept;
     }
 
-    pub fn start_listening_tcp(&mut self, port: u16) {
+    pub fn start_listening(&mut self) {
```

Comment:
Crust itself picks up listener addresses (with ports) from config. Since this mock doesn't actually listen for any networking connections, seems like this static port makes no difference.